src/Database/QueryCompiler.php

```diff
@@ -211,9 +212,15 @@ protected function _buildJoinPart($parts, $query, $generator)
     {
         $joins = '';
         foreach ($parts as $join) {
+            $subquery = $join['table'] instanceof Query || $join['table'] instanceof QueryExpression;
             if ($join['table'] instanceof ExpressionInterface) {
-                $join['table'] = '(' . $join['table']->sql($generator) . ')';
+                $join['table'] = $join['table']->sql($generator);
```

Comment:
I understand why it needs to change for your case, but will it break passing a Query in the from?